src/store/portfolioSummary/index.ts

```diff
@@ -0,0 +1,258 @@
+    portfolioSummaryList.forEach(portfolio => {
+      portfolio.task_orders.forEach(taskOrder => {
+        taskOrder.clin_records?.forEach(clinRecord => {
+          clinNumbers.push(clinRecord.clin_number);
+        })
+      })
+    });
+    const allCostList = await api.costsTable.getQuery(
+      {
+        params:
+          {
+            sysparm_fields: "sys_id,clin,task_order_number,is_actual,value",
+            sysparm_query: "clinIN" + clinNumbers
+          }
+      }
+    )
+    portfolioSummaryList.forEach(portfolio => {
+      portfolio.task_orders.forEach(taskOrder => {
+        taskOrder.clin_records?.forEach(clinRecord => {
+          clinRecord.cost_records =
+            allCostList.filter(cost => {
+              const clinNumber = cost.clin as unknown as string;
+              return clinNumber === clinRecord.clin_number &&
+              cost.task_order_number === taskOrder.task_order_number
+            }); // FIXME temp code above
+          // allCostList.filter(cost => cost.clin?.value === clinRecord.sys_id);//FIXME correct code
+        })
+      })
+    })
+  }
+
+  /**
+   * Performs the 'Total Obligated' and 'Funds Spent' for all the portfolios in the list
+   *
+   * NOTE: Instead of computing here if we were to make the aggregate API call, it would take
+   * as many calls as the number of portfolios to get, for example, the 'Total Obligated' from
+   * the CLINS table. The reason for this is, the aggregate query response does not have any
+   * reference to what portfolio the aggregate belongs to and so a call for each portfolio.
+   * Computing here will eliminate all these calls.
+   */
+  @Action({rawError: true})
+  private computeAllAggregations(portfolioSummaryList: PortfolioSummaryDTO[]) {
+    portfolioSummaryList.forEach(portfolio => {
+      portfolio.dod_component = 'ARMY' // FIXME: delete this line after API starts returning
+      let totalObligatedForPortfolio = 0;
+      let totalFundsSpentForPortfolio = 0;
+      portfolio.task_orders.forEach(taskOrder => {
+        taskOrder.clin_records?.forEach(clinRecord => {
+          if (clinRecord.clin_status === 'ACTIVE' ||
+            clinRecord.clin_status === 'OPTION_EXERCISED') { // TODO: double check the statuses
+            totalObligatedForPortfolio =
+              totalObligatedForPortfolio + Number(clinRecord.funds_obligated);
+          }
+          clinRecord.cost_records?.forEach(costRecord => {
+            totalFundsSpentForPortfolio = totalFundsSpentForPortfolio + Number(costRecord.value);
+          })
+        })
+      })
+      portfolio.funds_obligated = totalObligatedForPortfolio;
+      portfolio.funds_spent = totalFundsSpentForPortfolio;
+    })
+  }
+
+  @Action({rawError: true})
+  public async loadPortfolioSummaryList(): Promise<PortfolioSummaryDTO[]> {
+    await this.ensureInitialized();
+    try {
+      const query =
+        "portfolio_managersLIKEe0c4c728875ed510ec3b777acebb356"; // TODO get session of user
```

Comment:
failed secret scan in sonar cloud, but this is not a secret so you can add the comment `// pragma: allowlist secret` at the end of the line
![Screen Shot 2022-10-05 at 7 54 01 AM](https://user-images.githubusercontent.com/90333349/194055067-b549bd2d-1239-4e20-831b-68b9e5f831ba.png)

Reply:
Done!